src/routes/catalog/+page.svelte

```diff
@@ -0,0 +1,94 @@
+<script>
+    import { writable} from 'svelte/store'
+	import { getLoggedInUser } from '../../database/auth';
+    import { getRandomFolder, getImageUrlFromFolderRef, getChatName } from '../../database/catalog';
+	import { onMount } from 'svelte';
+    import { db } from '../../database/config';
+    import {
+	collection,
+	query,
+	where,
+	getDocs,
+} from 'firebase/firestore';
+
+const chatRoomsCollection = collection(db, 'chatRooms');
+    let user = null;
+    let imageUrls = writable([]); // Array to store image URLs
+
+
+    onMount(async () => {
+        user = await getLoggedInUser();
+        if (user) {
+            const userChatsQuery = query(chatRoomsCollection, where('members', 'array-contains', user.id));
+            const userChats = await getDocs(userChatsQuery);
+            const urls = [];
+
+            for (const chat of userChats.docs) {
+                const folderRef = await getRandomFolder(chat.id);
+                const chatName = await getChatName(chat.id)
+                if (folderRef) {
+                    const url = await getImageUrlFromFolderRef(folderRef);
+                    if (url) {
+                        urls.push({ imageUrl: url, chatName: chatName, chatId : chat.id }); // Add URL to the temporary array
+                    }
+                }
+            }
+
+            // Update the writable store with the array of image URLs
+            imageUrls.set(urls);
+        }
+    });
+
+    
+</script>
+<meta name="viewport" content="width=device-width, initial-scale=1.0">
+<div class="grid-container">
+    {#each $imageUrls as imageObj}
+    <div>
+        <figure>
+            <a href="catalog/{imageObj.chatId}" target="_blank">
+            <div class="image-wrapper">
+            <img src="{imageObj.imageUrl}" alt="{imageObj.chatId}"/></div></a>
+            <a href="catalog/{imageObj.chatId}" target="_blank">
+            <!-- svelte-ignore a11y-structure -->
+            <figcaption>{imageObj.chatName}</figcaption></a>
```

Comment:
maybe you could style this so that the text is not blue but has color: whitesmoke (or the main colour if we decide to remove the backgroud colour)